### PR TITLE
Added support for configuring TLS cert/key/root from ByteBuffers

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/software/amazon/awssdk/crt/io/TlsContextOptions.java [hunk 1]
```diff
@@ -136,6 +139,17 @@ public void initMTLSFromPath(String certificatePath, String privateKeyPath) {
         tlsContextOptionsInitMTLSFromPath(getNativeHandle(), certificatePath, privateKeyPath);
     }
 
+    /**
+     * Sets the path to the certificate that identifies this TLS host. Must be in
```
Comment (bretambrose) [inline src/main/java/software/amazon/awssdk/crt/io/TlsContextOptions.java]: Comment seems off here

File: src/main/java/software/amazon/awssdk/crt/io/TlsContextOptions.java [hunk 2]
```diff
@@ -136,6 +139,17 @@ public void initMTLSFromPath(String certificatePath, String privateKeyPath) {
...
+     * PEM format.
+     * 
+     * @param certificate PEM format certificate
+     * @param privateKey  PEM format private key
+     */
+    public void initMTLS(ByteBuffer certificate, ByteBuffer privateKey) {
```
Comment (bretambrose) [inline src/main/java/software/amazon/awssdk/crt/io/TlsContextOptions.java]: I like the "FromX" Suffix on these readability wise.  Maybe FromBuffer(s)?

Comment (bretambrose) [reply src/main/java/software/amazon/awssdk/crt/io/TlsContextOptions.java]: On second thought it seems like this mirrors the aws-c-io functions.  Maybe not worth changing.

File: src/main/java/software/amazon/awssdk/crt/io/TlsContextOptions.java [hunk 3]
```diff
@@ -16,6 +16,7 @@
 
 import software.amazon.awssdk.crt.CrtResource;
 import software.amazon.awssdk.crt.CrtRuntimeException;
+import software.amazon.awssdk.crt.utils.PemUtils;
```
Comment (ColdenCullen) [inline src/main/java/software/amazon/awssdk/crt/io/TlsContextOptions.java]: Homie you forgot to commit this file

File: src/test/java/software/amazon/awssdk/crt/test/TlsContextOptionsTest.java [hunk 4]
```diff
@@ -5,18 +5,95 @@
...
+            + "1yQbPnoKZrB7SopAkcVqZv4ks0LLu+BLfEFXYy8CgYEApN8xXDgoRVnCqQpN53iM\n"
+            + "n/c0iqjOXkTIb/jIksAdv3AAjaayP2JaOXul7RL2fJeshYiw684vbb/RNK6jJDlM\n"
+            + "sH0Pt6t3tZmB2bC1KFfh7+BMdjg/p63LC6PAasa3GanObh67YADPOfoghCsOcgzd\n"
+            + "6brt56fRDdHgE2P75ER/zm8CgYEArAxx6bepT3syIWiYww3itYBJofS26zP9++Zs\n"
+            + "T9rX5hT5IbMo5vwIJqO0+mDVrwQfu9Wc7vnwjhm+pEy4qfPW6Hn7SNppxnY6itZo\n"
+            + "J4/azOIeaM92B5h3Pv0gxBFK8YyjO8beXurx+79ENuOtfFxd8knOe/Mplcnpurjt\n"
+            + "SeVJuG8CgYBxEYouOM9UuZlblXQXfudTWWf+x5CEWxyJgKaktHEh3iees1gB7ZPb\n"
+            + "OewLa8AYVjqbNgS/r/aUFjpBbCov8ICxcy86SuGda10LDFX83sbyMm8XhktfyC3L\n"
+            + "54irVW5mNUDcA8s9+DloeTlUlJIr8J/RADC9rpqHLaZzcdvpIMhVsw==\n" 
+            + "-----END RSA PRIVATE KEY-----";
```
Comment (ColdenCullen) [inline src/test/java/software/amazon/awssdk/crt/test/TlsContextOptionsTest.java]: Please delete my private key so I don't get hacked

File: src/main/java/software/amazon/awssdk/crt/utils/PemUtils.java [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+     */
+    public static String removeInvalidPemChars(String pem) {
+        if (pem == null || pem.length() == 0) {
+            return pem;
+        }
+        return pem.replaceAll(UNNECESSARY_PEM_CHARACTER_SET, "");
+    }
+
+    /**
+     * Removes characters that are not valid in PEM format. No other cleanup is
```
Comment (alexw91) [inline src/main/java/software/amazon/awssdk/crt/utils/PemUtils.java]: "not valid Base64 characters"

File: src/main/java/software/amazon/awssdk/crt/io/TlsContextOptions.java [hunk 6]
```diff
@@ -136,6 +137,24 @@ public void initMTLSFromPath(String certificatePath, String privateKeyPath) {
...
+    /**
+     * Sets the certificate/key pair that identifies this TLS host. Must be in
+     * PEM format.
+     * 
+     * @param certificate PEM armored certificate
+     * @param privateKey  PEM armored private key
+     * @throws IllegalArgumentException If the certificate or privateKey are not in PEM format or if they contain chains
+     */
+    public void initMTLS(String certificate, String privateKey) throws IllegalArgumentException {
+        certificate = PemUtils.removeInvalidPemChars(certificate);
```
Comment (alexw91) [inline src/main/java/software/amazon/awssdk/crt/io/TlsContextOptions.java]: `cleanUpPem()` already calls `removeInvalidPemChars()`, so I don't think you need to call it here.

File: src/main/java/software/amazon/awssdk/crt/io/TlsContextOptions.java [hunk 7]
```diff
@@ -136,6 +137,24 @@ public void initMTLSFromPath(String certificatePath, String privateKeyPath) {
...
+        certificate = PemUtils.cleanUpPem(certificate);
```
Comment (alexw91) [inline src/main/java/software/amazon/awssdk/crt/io/TlsContextOptions.java]: If you're cleaning up PEM's be default, I think there should be a check to see if the PEM changed after being cleaned up, and a warning logged if it did change.

File: src/main/java/software/amazon/awssdk/crt/utils/PemUtils.java [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+            // allows us to reference
+            // Group 1 with "$1" in the replaceAll() call here.
+            // For more info see: https://stackoverflow.com/a/27328750/7565918
+            return m.replaceAll("-----END $1-----\n-----BEGIN");
+        }
+        return pem;
+    }
+
+    /**
+     * Merge consecutive spaces into a single space (Eg "BEGIN CERTIFICATE", will
```
Comment (alexw91) [inline src/main/java/software/amazon/awssdk/crt/utils/PemUtils.java]: Comment corrected a purposeful typo: 
```
"BEGIN    CERTIFICATE", will become "BEGIN CERTIFICATE"
```

**Comment by justinboswell** [comment]
I'll do a follow up and address these, thanks @alexw91 !

**Comment by justinboswell** [comment]
https://github.com/awslabs/aws-crt-java/pull/99 cc @alexw91

**Pull request merged**